src/components/application_manager/src/rpc_passing_handler.cc

```diff
@@ -254,34 +254,73 @@ bool RPCPassingHandler::CycleThroughRPCPassingRequests(
 void RPCPassingHandler::OnPassThroughRequestTimeout() {
   LOG4CXX_DEBUG(logger_, "RPC Passing request timeout");
   auto timeout_entry = timeout_queue_.front();
-  uint32_t correlation_id = std::get<1>(timeout_entry);
-  timeout_queue_.pop();
+  uint32_t correlation_id = timeout_entry.second;
+  RemoveTimeoutTimer(correlation_id);
   LOG4CXX_DEBUG(logger_,
                 "Cycling through requests for correlation_id "
                     << correlation_id);
   CycleThroughRPCPassingRequests(correlation_id);
+  if (request_queue_[correlation_id].empty()) {
+    LOG4CXX_DEBUG(logger_,
+                  "No services left in map. using core to handle request");
+    current_request_[correlation_id]
+        .message[strings::params][strings::connection_key] =
+        current_request_[correlation_id].origin_connection_key;
+    smart_objects::SmartObjectSPtr result =
+        std::make_shared<smart_objects::SmartObject>(
+            current_request_[correlation_id].message);
+    request_queue_.erase(correlation_id);
+    current_request_.erase(correlation_id);
+    messages_using_core_.insert(correlation_id);
+    RemoveTimeoutTimer(correlation_id);
```

Comment:
Seems that the `RemoveTimeoutTimer` above (https://github.com/smartdevicelink/sdl_core/pull/2821/commits/93abd286e45d3afdb5f9aa3178b63fe5072be5b6#diff-4d2e0b56e9fb5e814e5732bef1bd78a1R258) would already have taken care of this

Reply:
Fixed 👍